### PR TITLE
Support Autofac6

### LINKED ISSUE #39: Autofac 6 migration
Hi,

The Autofac team did a lot of fundametal changes in version 6 [(what's new)](https://autofaccn.readthedocs.io/en/latest/whats-new/upgradingfrom5to6.html).
Specifically:
>RegistrationBuilder.RegistrationData no longer exposes the configured ActivatedHandlers, ActivatingHandlers or PreparingHandlers, and IComponentRegistration no longer exposes Preparing, Activating or Activated events. 

And current "autofac-serilog-integration" version (4.0) throw an exception, because they've removed a couple of used by you methods.
Are you going to fix it soon ?

Thanks,

### PR DESCRIPTION
**Pull request by krajek**
Fixes #39.
Please review the code since this is the first time I worked with the internals of Autofac :-).
Commented out code is for now left, for easier reference.

### CODE REVIEW

**Comment by krajek** [comment]
FYI: I have run this version of the package on my semi-large project and it worked perfectly.

File: src/AutofacSerilogIntegration/AutofacSerilogIntegration.csproj [hunk 1]
```diff
@@ -1,9 +1,9 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <Description>Contextual logger injection for Autofac using Serilog</Description>
     <AssemblyTitle>AutofacSerilogIntegration</AssemblyTitle>
-    <VersionPrefix>4.0.1</VersionPrefix>
+    <VersionPrefix>5.0.0-alpha2</VersionPrefix>
```
Comment (nblumhardt) [inline src/AutofacSerilogIntegration/AutofacSerilogIntegration.csproj:6]: The `-alpha2` suffix isn't needed here, prerelease tags are added automatically by the build

File: src/AutofacSerilogIntegration/ContextualLoggingModule.cs [hunk 2]
```diff
@@ -44,45 +44,55 @@ protected override void Load(ContainerBuilder builder)
...
-                builder.Register(c =>
+                RegisterLoggerAndProviderForDisposingMode(builder);
+            }
+            else
+            {
+                RegisterLoggerForNonDisposingMode(builder);
+            }
+        }
+
+        private void RegisterLoggerAndProviderForDisposingMode(ContainerBuilder builder)
```
Comment (nblumhardt) [inline src/AutofacSerilogIntegration/ContextualLoggingModule.cs:55]: Nit: code doesn't use explicit `private` or `internal`

**Review by nblumhardt — COMMENTED** [review]
LGTM, just some minor nitpicks :+1:

File: src/AutofacSerilogIntegration/AutofacSerilogIntegration.csproj (hunk 1 above)
Comment (krajek) [reply src/AutofacSerilogIntegration/AutofacSerilogIntegration.csproj]: Fixed.

File: src/AutofacSerilogIntegration/ContextualLoggingModule.cs (hunk 2 above)
Comment (krajek) [reply src/AutofacSerilogIntegration/ContextualLoggingModule.cs]: Fixed.

**Comment by krajek** [comment]
Hi @nblumhardt , thanks for the review, code is fixed.

Please merge #41 before this one if possible.

Additionally, should I adjust the readme, to make sure consumers understand the implications of updating `AutofacSerilogIntegration`?

**Comment by nblumhardt** [comment]
@krajek thanks! I think supporting the latest Autofac with a major version bump in this package will be unsurprising, so let's leave the README for now. The package itself documents its dependencies accurately.

**Pull request merged**